Userland/Libraries/LibJS/Runtime/Array.cpp

```diff
@@ -23,16 +23,27 @@ Array* Array::create(GlobalObject& global_object, size_t length, Object* prototy
     if (!prototype)
         prototype = global_object.array_prototype();
     auto* array = global_object.heap().allocate<Array>(global_object, *prototype);
-    array->put(vm.names.length, Value(length));
+    array->internal_define_own_property(vm.names.length, { .value = Value(length), .writable = true, .enumerable = false, .configurable = false });
     return array;
 }
 
 // 7.3.17 CreateArrayFromList ( elements ), https://tc39.es/ecma262/#sec-createarrayfromlist
-Array* Array::create_from(GlobalObject& global_object, const Vector<Value>& elements)
+Array* Array::create_from(GlobalObject& global_object, Vector<Value> const& elements)
 {
+    // 1. Assert: elements is a List whose elements are all ECMAScript language values.
+
+    // 2. Let array be ! ArrayCreate(0).
```

Comment:
Not that it's a bad thing, but any reason for the very verbose spec copy here?

Reply:
Multiple reasons:

- you can pretty much immediately see if a function is fully or partially spec compliant - if it has these comments everywhere I'm trusting it, and if it doesn't I'm much more sceptical if it really does what it should do
- it encourages spec compliant implementations of new code (similar to adding AO names and links - it's much harder to half-ass something if a comment one line above tells a different story)
- it's easier to see when the code gets out of sync with the spec, which it likely will at some point - the spec evolves (other engines simply use `// 1.`, `// 2.` for that, but if we already add comments we might as well add the text, too)
- easier to review, maybe?

It's quite verbose, I know, and I'm not expecting anyone else to do this (kling already did) - but removing it is not up for debate :)

Reply:
Alright, if that's something you want to do, then go for it.